components/ILIAS/ILIASObject/classes/class.ilObjectGUI.php

```diff
@@ -953,13 +953,23 @@ public function saveObject(): void
 
         $class_name = 'ilObj' . $this->obj_definition->getClassName($this->requested_new_type);
 
+        /** @var ilObjectPropertyTitleAndDescription $title_and_description_prop */
+        $title_and_description_prop = $data['title_and_description'];
+
         $new_obj = new $class_name();
         $new_obj->setType($this->requested_new_type);
         $new_obj->processAutoRating();
+        /*
+         * Title and description needs to be set before calling ilObject::create
+         * at least preliminarily, because the creation of a new LOM set requires
+         * the object to have a non-null title.
+         */
+        $new_obj->setTitle($title_and_description_prop->getTitle());
```

Comment:
Hi @schmitz-ilias 

Thank you for the PR. This is not a review yet, even if the changes look ok on a first glance. Just one reaction, as I think this is something bigger: I don't think setting the title and description here works. This is done directly in the ObjectProperties. Currently there are no `ObjectProperties`, before an object is created in the database, so this should go sideways. I did not test this, but the code and my experience suggests it. This has to change one day, but I don't have financing to change this right now and it is more then just a couple of minutes.

Thanks again and best,
@kergomard

Reply:
Hi @kergomard 

Thanks for the quick feedback. Setting title and description here via `ilObjectProperties::storePropertyTitleAndDescription` would indeed go sideways, but I'm just setting title and description in the `ilObject` itself as local properties, nothing is persisted. This is needed, because that is where `ilObject::createMetadata` grabs title and description from.

As it's currently implemented, one could also just set the title to some dummy value, any non-empty string would do. It is overwritten again a few lines lower anyways, that time actually via `ilObjectProperties`. Using the raw title and description entered by the user seemed more robust to me.

For what it's worth, I tested the creation of objects on trunk at least for Courses and Groups, and that seems to work fine. As you already said, this all should be untangled properly at some point, but for now this seems like the slimmest solution.